CryptomatorCommon/Sources/CryptomatorCommonCore/WebDAV/WebDAVAuthenticator+Keychain.swift

```diff
@@ -9,15 +9,32 @@
 import CryptomatorCloudAccessCore
 import Foundation
 
+public enum WebDAVAuthenticatorKeychainError: Error {
+	case credentialDuplicate(existingIdentifier: String)
+}
+
 public extension WebDAVAuthenticator {
 	static func getCredentialFromKeychain(with accountUID: String) -> WebDAVCredential? {
 		return CryptomatorKeychain.webDAV.get(accountUID)
 	}
 
-	static func saveCredentialToKeychain(_ credential: WebDAVCredential, with accountUID: String) throws {
+	/**
+	 Saves a WebDAV credential to the keychain.
+
+	 Checks for duplicates before saving the passed credential to the keychain.
+	 A duplicate is defined as any other WebDAV credential with the same `baseURL` and `username`.
+	 - Throws: An WebDAVAuthenticatorKeychainError.credentialDuplicate if a WebDAVCredential already exists in the keychain with the same `baseURL` and `username` but a different identifier.
+	 The error includes the identifier (`existingIdentifier`) of the WebDAVCredentials item, which is already stored in the keychain and caused the error.
+	 */
+	static func saveCredentialToKeychain(_ credential: WebDAVCredential) throws {
+		let existingCredentials = try CryptomatorKeychain.webDAV.getAllWebDAVCredentials()
+		if let existingCredential = existingCredentials.first(where: { $0.baseURL == credential.baseURL && $0.username == credential.username && $0.identifier != credential.identifier }) {
```

Comment:
This obviously works but I'm curious what you think about making `WebDAVCredential` conform to `Equatable` with a custom implementation of `static func ==(...)`.